lib/buffer.js

```diff
@@ -488,6 +488,59 @@ Buffer.concat = function concat(list, length) {
   return buffer;
 };
 
+function fastcopy(source, target, targetStart = 0, sourceStart = 0, sourceEnd) {
+  if (!isUint8Array(source)) {
+    throw new ERR_INVALID_ARG_TYPE('source', ['Buffer', 'Uint8Array'], source);
+  }
+
+  if (!isUint8Array(target)) {
+    throw new ERR_INVALID_ARG_TYPE('target', ['Buffer', 'Uint8Array'], target);
+  }
+
+  if (targetStart < 0)
+    throw new ERR_OUT_OF_RANGE('targetStart', '>= 0', targetStart);
+  else
+    targetStart >>>= 0;
+
+  if (sourceStart < 0)
+    throw new ERR_OUT_OF_RANGE('sourceStart', '>= 0', sourceStart);
+  else
+    sourceStart >>>= 0;
+
+  if (sourceEnd === undefined)
+    sourceEnd = source.byteLength;
+  else if (sourceEnd < 0)
+    throw new ERR_OUT_OF_RANGE('sourceEnd', '>= 0', sourceEnd);
+  else
+    sourceEnd >>>= 0;
+
+  if (targetStart >= target.byteLength || sourceStart >= sourceEnd)
+    return 0;
+
+  if (sourceStart > source.byteLength) {
+    throw new ERR_OUT_OF_RANGE(
+      'sourceStart', `<= ${source.byteLength}`, sourceStart
+    );
+  }
+
+  if (sourceEnd - sourceStart > target.byteLength - targetStart)
+    sourceEnd = sourceStart + target.byteLength - targetStart;
```

Comment:
Please add a comment what this does (if I read it correct, it's the overflow mechanism to search from the end instead of the beginning).